android/app/src/main/java/org/smartregister/fhircore/activity/CqlActivity.kt

```diff
@@ -0,0 +1,194 @@
+            ancLibrary to fhirHelpersLibrary
+          }
+        }
+
+      val libraryStream = fhirContext.newJsonParser().encodeResourceToString(ancLibrary)
+      val fhirHelpersStream = fhirContext.newJsonParser().encodeResourceToString(fhirHelperLibrary)
+      val library = parser.parseResource(libraryStream)
+      val fhirHelpersLibrary = parser.parseResource(fhirHelpersStream)
+      val resources: List<IBaseResource> = Lists.newArrayList(library, fhirHelpersLibrary)
+      val bundleFactory = fhirContext.newBundleFactory()
+      val bundleLinks = BundleLinks("", null, true, BundleTypeEnum.COLLECTION)
+      bundleFactory.addRootPropertiesToBundle(
+        "bundled-directory",
+        bundleLinks,
+        resources.size,
+        null
+      )
+      bundleFactory.addResourcesToBundle(
+        resources,
+        BundleTypeEnum.COLLECTION,
+        "",
+        BundleInclusionRule.BASED_ON_INCLUDES,
+        null
+      )
+      contentProvider =
+        BundleFhirLibraryContentProvider(
+          fhirContext,
+          bundleFactory.resourceBundle as IBaseBundle,
+          adapterFactory,
+          libraryVersionSelector
+        )
+
+      // Load terminology content, and create a TerminologyProvider which is the interface used by
+      // the evaluator for resolving terminology
+      val valueSetStream = this.assets.open("valueset/valueset-bundle.json")
+      val valueSetBundle = parser.parseResource(valueSetStream)
+      terminologyProvider = BundleTerminologyProvider(fhirContext, valueSetBundle as IBaseBundle)
+
+      // Load data content, and create a RetrieveProvider which is the interface used for
+      // implementations of CQL retrieves.
+      val dataStream = this.assets.open("test/mom-with-anaemia-bundle.json")
+      val dataBundle = parser.parseResource(dataStream)
+      bundleRetrieveProvider = BundleRetrieveProvider(fhirContext, dataBundle as IBaseBundle)
+      bundleRetrieveProvider!!.terminologyProvider = terminologyProvider
+      bundleRetrieveProvider!!.isExpandValueSets = true
+    } catch (e: IOException) {
+      e.printStackTrace()
+      throw RuntimeException(e)
+    }
+    cqlEvaluator =
+      CqlEvaluator(
+        object :
+          TranslatingLibraryLoader(
+            ModelManager(),
+            listOf(contentProvider),
+            CqlTranslatorOptions.defaultOptions()
+          ) {
+          // This is a hack needed to circumvent a bug that's currently present in the cql-engine.
+          // By default, the LibraryLoader checks to ensure that the same translator options are
+          // used to for all libraries,
+          // And it will re-translate if possible. Since translating CQL is not currently possible
+          // on Android (some changes to the way ModelInfos are loaded is needed) the library loader
+          // just needs to load libraries
+          // regardless of whether the options match.
```

Comment:
what are the situations where the options don't match?

Reply:
haven't considered about such situations yet

Reply:
k so then do we need this hack? b/c if we don't have those situations I don't see why we'd need it

Reply:
yes, you are right

Reply:
ok cool, in that case, can we remove this section of code?